components/compositing/compositor.rs

```diff
@@ -478,7 +478,8 @@ impl<Window: WindowMethods> IOCompositor<Window> {
             }
 
             (Msg::CreatePng(reply), ShutdownState::NotShuttingDown) => {
-                let img = self.composite_specific_target(CompositeTarget::WindowAndPng);
+                let res = self.composite_specific_target(CompositeTarget::WindowAndPng);
+                let img = res.ok().and_then(|x| x);
```

Comment:
But I guess I'm looking for `unwrap_or`. The `Result` and `Option` APIs are huge :(